### PR TITLE
Signals UI validation

### PR DESCRIPTION
**Pull request by tony-tvu**
**Description:**
Implement Validation for Email Addresses and Webhook URLs in Signal Forms

**Issue:**
https://github.com/target/goalert/issues/3406

**Out of Scope:**
Resolved issue: Form Availability Consistency – Now, querying all Slack channels no longer triggers errors, ensuring uninterrupted access to the form for all users.

### CODE REVIEW

File: web/src/app/services/ServiceRules/ServiceRuleCreateDialog.tsx [hunk 1]
```diff
@@ -158,12 +157,11 @@ export default function ServiceRuleCreateDialog(props: {
   const [actionsError, setActionsError] = useState<boolean>(false)
   const [createRuleStatus, commit] = useMutation(mutation)
 
-  const [{ fetching, error, data }] = useQuery({
+  const [{ fetching, data }] = useQuery({
     query,
     variables: {},
   })
   if (fetching && !data) return <Spinner />
-  if (error) return <GenericError error={error.message} />
```
Comment (Forfold) [inline web/src/app/services/ServiceRules/ServiceRuleCreateDialog.tsx:166]: We could perhaps pass this error to the FormDialog so this doesn't get missed in any edge cases:
`errors={nonFieldErrors(createRuleStatus.error || error)}`

File: web/src/app/services/ServiceRules/ServiceRuleForm.tsx [hunk 2]
```diff
@@ -300,6 +300,53 @@ export default function ServiceRuleForm(
...
+        (action.destType === destType.SLACK && c.prop === 'channel') ||
+        (action.destType === destType.SLACK && c.prop === 'channel_id')
+      ) {
+        return <React.Fragment key={contentIdx} />
+      }
+      return (
+        <Grid
+          key={c.prop}
+          item
+          style={{ flexGrow: 1, marginTop: '1em' }}
```
Comment (Forfold) [inline web/src/app/services/ServiceRules/ServiceRuleForm.tsx:318]: nit: use the sx prop on MUI components moving forward. It should work the same but uses mui's newer `@emotion` code instead
```suggestion
          sx={{ flexGrow: 1, marginTop: '1em' }}
```

**Review by Forfold — CHANGES_REQUESTED** [review]

File: web/src/app/services/ServiceRules/ServiceRuleForm.tsx [hunk 3]
```diff
@@ -300,6 +300,48 @@ export default function ServiceRuleForm(
...
+  const renderContents = (
+    actionIdx: number,
+    action: ServiceRuleActionInput,
+  ): React.ReactNode => {
+    return action.contents.map((c: Content, contentIdx: number) => {
+      if (
+        (action.destType === destType.SLACK && c.prop === 'channel') ||
+        (action.destType === destType.SLACK && c.prop === 'channel_id')
+      ) {
+        return <React.Fragment key={contentIdx} />
```
Comment (Forfold) [inline web/src/app/services/ServiceRules/ServiceRuleForm.tsx:312]: let's supply the key with some additional info on top of the index, to ensure each key is always unique and can be properly referenced:

```suggestion
        return <React.Fragment key={contentIdx+c.prop} />
```

Comment (tony-tvu) [reply web/src/app/services/ServiceRules/ServiceRuleForm.tsx]: I went back and did this for all service rule files that are using a key :thumbsup:.

**Comment by tony-tvu** [comment]
Closing since signals UI destinations implementation from this PR is much different than how destinations is currently working on master branch today.

**Pull request closed without merging**